lib/pages/goal_page.dart

```diff
@@ -3,102 +3,529 @@ import 'package:linear/model/goal.dart';
+            i--;
+           });
+       }
+     }
+     setState(() {
+    _isSearching = false;
+  });
+ }
+
+findSearchResults(){
+  String community = "";
+   if(searchInput.text.isEmpty) {
+      return;
+    }else{
+      community = searchInput.text.toString().toLowerCase();
+    }
+  setState(() {
+    _goals = _allGoals.toList();
+    _isSearching = true;
+    isSorting = false;
+  });
+   Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.communityName) != community){
+            setState(() {
+             _goals.removeAt(i);
+            i--;
+           });
+       }
+     }
+     setState(() {
+    _isSearching = false;
+  });
+ }
+
+ getAll(List getGoals){
+  setState(() {
+    _goals = _allGoals.toList();
+  });
+ }
+
+ sortByNewest(List sortGoals){
+  sortGoals.sort((a, b) => b['creationDate'].compareTo(a['creationDate']));
+    setState(() {
+      _goals = sortGoals.toList();
+   });
+ }
+
+ sortByOldest(List sortGoals){
+  sortGoals.sort((a, b) => a['creationDate'].compareTo(b['creationDate']));
+  setState(() {
+      _goals = sortGoals.toList();
+   });
+ }
+
+ sortByMostDone(List sortGoals){
+  sortGoals.sort((a, b) => ((b['completedCheckIns']/b['checkInGoal'])).compareTo((a['completedCheckIns']/a['checkInGoal']))); 
+  setState(() {
+      _goals = sortGoals.toList();
+   });
+ }
+
+ sortByLeastDone(List sortGoals){
+  sortGoals.sort((a, b) => ((a['completedCheckIns']/a['checkInGoal'])).compareTo((b['completedCheckIns']/b['checkInGoal']))); 
+  setState(() {
+      _goals = sortGoals.toList();
+   });
+ }
+
+ sortByCommunityAZ(List sortGoals){
+  sortGoals.sort((a, b) => a['community'].compareTo(b['community'])); 
+  setState(() {
+      _goals = sortGoals.toList();
+   });
+ }
+
+ sortByCommunityZA(List sortGoals){
+  sortGoals.sort((a, b) => b['community'].compareTo(a['community'])); 
+  setState(() {
+      _goals = sortGoals.toList();
+   });
+ }
+
+ filterMenu(int value){
+   if (value == 0) {
+      filteringBy = "Open";
+      getOpen(_goals);
+    } else if (value == 1) {
+      filteringBy = "Overdue";
+      getOverdue(_goals);
+    } else if (value == 2) {
+      sortingBy = "Finished";
+      getFinished(_goals);
+    } else if (value == 3) {
+      sortingBy = "All";
+      getAll(_goals);
+    } 
+ }
+
+ sortMenu(int value){
+    if (value == 0) {
+      sortingBy = "Newest";
+      sortByNewest(_goals);
+    } else if (value == 1) {
+      sortingBy = "Oldest";
+      sortByOldest(_goals);
+    } else if (value == 2) {
+      sortingBy = "Completion↑";
+      sortByMostDone(_goals);
+    } else if (value == 3) {
+      sortingBy = "Completion↓";
+      sortByLeastDone(_goals);
+    } else if (value == 4){
+      sortingBy = "Communities A to Z";
+      sortByCommunityAZ(_goals);
+    }else if (value == 5) {
+      sortingBy = "Communities Z to A";
+      sortByCommunityZA(_goals);
+    }
+ }
```

Comment:
we won't need these with the `applySort()` and `applyFilter()` functions mentioned above.

we'd replace them with:

```
setState(() {
   sortingBy = SortingField.NEWEST
});
applySort()
```

and same for filter